example/authserver/main.go

```diff
@@ -77,47 +75,50 @@ type TestUserVerifier struct {
 }
 
 // ValidateUser validates username and password returning an error if the user credentials are wrong
-func (*TestUserVerifier) ValidateUser(username, password, scope string, req *http.Request) error {
+func (*TestUserVerifier) ValidateUser(username, password, scope string, r *http.Request) error {
 	if username == "user01" && password == "12345" {
 		return nil
 	}
-	return errors.New("Wrong user")
+
+	return errors.New("wrong user")
 }
 
-// ValidateClient validates clientId and secret returning an error if the client credentials are wrong
-func (*TestUserVerifier) ValidateClient(clientID, clientSecret, scope string, req *http.Request) error {
+// ValidateClient validates clientID and secret returning an error if the client credentials are wrong
+func (*TestUserVerifier) ValidateClient(clientID, clientSecret, scope string, r *http.Request) error {
 	if clientID == "abcdef" && clientSecret == "12345" {
 		return nil
 	}
-	return errors.New("Wrong client")
+
+	return errors.New("wrong client")
+}
+
+// ValidateCode validates token ID
+func (*TestUserVerifier) ValidateCode(clientID, clientSecret, code, redirectURI string, r *http.Request) (string, error) {
+	return "", nil
 }
 
 // AddClaims provides additional claims to the token
-func (*TestUserVerifier) AddClaims(ctx context.Context, tokenType oauth.TokenType, credential, tokenID, scope string) (map[string]string, error) {
+func (*TestUserVerifier) AddClaims(tokenType oauth.TokenType, credential, tokenID, scope string, r *http.Request) (map[string]string, error) {
 	claims := make(map[string]string)
-	claims["customerId"] = "1001"
-	claims["customerData"] = `{"OrderDate":"2016-12-14","OrderId":"9999"}`
+	claims["customer_id"] = "1001"
```

Comment:
In JS world / json, the convention is to use camelCase as it was before, but both are fine..

Reply:
Per the standard, JSON doesn't impose any convention on keys. I think the conventions are more determined by authors of the API.
Facebook and Twitter use snake case for their JSON in their APIs:
- https://developers.facebook.com/docs/graph-api/field-expansion/
- https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/overview
Google uses camel case:
- https://google.github.io/styleguide/jsoncstyleguide.xml

 I prefer snake case, but am happy to do whatever for these examples.

Reply:
Also, all of the oAuth structs for this package (and the gin version) use snake case. 🤷🏻

Reply:
@pkieltyka can we merge this or are you not going to until I update those JSON keys?